services/src/test/java/edu/unc/lib/dl/cdr/services/processing/AccessControlRetrievalServiceTest.java

```diff
@@ -0,0 +1,181 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.cdr.services.processing;
+
+import static edu.unc.lib.dl.acl.util.UserRole.canManage;
+import static edu.unc.lib.dl.acl.util.UserRole.canViewMetadata;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+import edu.unc.lib.dl.acl.fcrepo4.InheritedAclFactory;
+import edu.unc.lib.dl.acl.service.AccessControlService;
+import edu.unc.lib.dl.acl.service.PatronAccess;
+import edu.unc.lib.dl.acl.util.AgentPrincipals;
+import edu.unc.lib.dl.acl.util.UserRole;
+import edu.unc.lib.dl.fcrepo4.ContentContainerObject;
+import edu.unc.lib.dl.fcrepo4.ContentObject;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectLoader;
+import edu.unc.lib.dl.fedora.PID;
+
+/**
+ *
+ * @author lfarrell
+ * @author harring
+ *
+ */
+public class AccessControlRetrievalServiceTest {
+    private static final String MANAGE_PRINC = "manageGrp";
+    private static long ONE_DAY = 86400000;
+    private static final String PATRON_PRINC = "everyone";
+
+    @Mock
+    private RepositoryObjectLoader repoObjLoader;
+    @Mock
+    private ContentObject member;
+    @Mock
+    private ContentContainerObject parent;
+    @Mock
+    private InheritedAclFactory aclFactory;
+    @Mock
+    private AccessControlService aclService;
+    @Mock
+    private AgentPrincipals agent;
+    @Mock
+    private ArrayList<Map<String, Object>> memberPermissions;
+    @Mock
+    private  Map<String, Object> permissions;
+    @Mock
+    private Map<String, Object> permResult;
+
+    private PID parentPid;
+    private PID memberPid;
+    private String memberUuid;
+    private String parentUuid;
+    private Map<String, Object> result;
+    private AccessControlRetrievalService aclRetrievalService;
+    private Map<String, Set<String>> objPrincRoles;
+    private Date testDate;
+
+    @Before
+    public void init() {
+        initMocks(this);
+
+        parentPid = PIDs.get(UUID.randomUUID().toString());
+        memberPid = PIDs.get(UUID.randomUUID().toString());
+        memberUuid = memberPid.getId();
+        parentUuid = parentPid.getId();
+        testDate = new Date(System.currentTimeMillis() + ONE_DAY);
```

Comment:
what happened after that part?

Reply:
It went into a timezone fueled rage, even though its clearly "local" so you'd think it'd just go with the local system default unless otherwise prompted.  A little surprised it doesn't have a `.toDate()` method rather than making it multiple steps.  I guess I'm fine with it, some parts of it are very readable, but I guess I didn't feel compelled to switch everything over it that api.